pkg/resource/deploy/deployment_executor.go

```diff
@@ -233,6 +250,16 @@ func (ex *deploymentExecutor) Execute(callerCtx context.Context, opts Options, p
 				}
 
 				if event.Event == nil {
+					if workerPool != nil {
+						// Wait for all currently pending and executing workers to
+						// complete. Then check if any worker had an error.
+						poolErr := workerPool.Wait(true)
+						if poolErr != nil {
+							cancel()
+							return false, result.BailError(poolErr)
+						}
+					}
+
 					// Check targets before performDeletes mutates the initial Snapshot.
 					targetErr := ex.checkTargets(opts.Targets)
 
```

Comment:
`performDeletes` below also does step generation, do we want to parallelize that as well?

Reply:
probably not necessary. tbh the only reason that parallelizing same/update/replace step gen is useful is that it allows provider work (check, diff) to be done concurrently. IIRC delete step gen is all internal to the step generator and probably wouldn't benefit from parallelism

Reply:
I'll add a comment to make this clear